rebuild/src/functions.cc

```diff
@@ -0,0 +1,583 @@
+#include "functions.h"
+
+namespace {
+Napi::Value AggregatePublicKeys(const Napi::CallbackInfo &info) {
+    BLST_TS_FUNCTION_PREAMBLE
+    if (!info[0].IsArray()) {
+        Napi::TypeError::New(env, "publicKeys must be of type PublicKeyArg[]")
+            .ThrowAsJavaScriptException();
+        return scope.Escape(env.Undefined());
+    }
+    Napi::Array arr = info[0].As<Napi::Array>();
+    uint32_t length = arr.Length();
+    if (length == 0) {
+        Napi::TypeError::New(env, "PublicKeyArg[] must have length > 0")
+            .ThrowAsJavaScriptException();
+        return scope.Escape(env.Undefined());
+    }
+
+    BLST_TS_CREATE_JHEAP_OBJECT(wrapped, public_key, PublicKey, result)
+    result->_has_jacobian = true;
+    result->_jacobian.reset(new blst::P1);
+
+    for (uint32_t i = 0; i < length; i++) {
+        Napi::Value val = arr[i];
+        PointerGroup<blst::P1> ptr_group;
+        try {
+            if (unwrap_point_arg(
+                    ptr_group,
+                    env,
+                    module,
+                    val,
+                    "PublicKeyArg",
+                    CoordType::Jacobian)) {
+                return scope.Escape(env.Undefined());
+            }
+            // TODO: Do we still need to check for 0x40 key?
```

Comment:
I will add spec tests in a separate PR and will verify if this still needs to be checked or not.